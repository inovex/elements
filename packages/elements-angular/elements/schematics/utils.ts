import { virtualFs, workspaces } from '@angular-devkit/core';
import { SchematicsException, Tree } from '@angular-devkit/schematics';

const CONFIG_PATH = 'angular.json';

function createHost(tree: Tree): workspaces.WorkspaceHost {
  return {
    async readFile(path: string): Promise<string> {
      const data = tree.read(path);
      if (!data) {
        throw new SchematicsException('File not found.');
      }
      return virtualFs.fileBufferToString(data);
    },
    async writeFile(path: string, data: string): Promise<void> {
      return tree.overwrite(path, data);
    },
    async isDirectory(path: string): Promise<boolean> {
      return !tree.exists(path) && tree.getDir(path).subfiles.length > 0;
    },
    async isFile(path: string): Promise<boolean> {
      return tree.exists(path);
    },
  };
}

export function readConfig(host: Tree) {
  const sourceText = host.read(CONFIG_PATH)?.toString('utf-8') || '';
  return JSON.parse(sourceText);
}

export function writeConfig(host: Tree, config: JSON) {
  host.overwrite(CONFIG_PATH, JSON.stringify(config, null, 2));
}

export function getAngularAppConfig(
  config: any,
  projectName: string
): any | never {
  if (!config.projects.hasOwnProperty(projectName)) {
    throw new SchematicsException(`Could not find project: ${projectName}`);
  }

  return config.projects[projectName];
}

export async function getProjectName(tree: Tree) {
  const host = createHost(tree);
  const { workspace } = await workspaces.readWorkspace('/', host);
  return workspace.extensions.defaultProject as string;
}
