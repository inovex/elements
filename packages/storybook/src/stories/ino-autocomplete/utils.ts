import { KeyValue } from '@inovex.de/elements';

export const AutoCompleteStoryOptions: KeyValue[] = [
  {
    key: 'hh',
    value: 'Hamburg',
  },
  {
    key: 'ber',
    value: 'Berlin',
  },
  {
    key: 'k',
    value: 'Karlsruhe',
  },
];
