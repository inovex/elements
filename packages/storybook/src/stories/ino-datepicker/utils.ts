import '@inovex.de/elements';

export function registerInlineDatepickerHandler() {
  const group = document.querySelector('ino-radio-group') as HTMLInoRadioGroupElement;
  const datepicker = document.querySelector('.datepicker-group ino-datepicker') as HTMLInoDatepickerElement;

  const today = document.querySelector('.today');
  const thisWeek = document.querySelector('.thisWeek');
  const lastWeek = document.querySelector('.lastWeek');

  const todayClickHandler = () => {
    datepicker.value = '02.03.2020';
    datepicker.label = 'Am';
    group.value = 'at';
    datepicker.range = false;
  };

  const thisWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '02.03.2020 bis 08.03.2020';
  };

  const lastWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '24.02.2020 bis 01.03.2020';
  };

  const checkedChangeHandler = (e: CustomEvent<boolean>) => {
    const target = e.target as HTMLInoRadioElement;

    switch (target.value) {
      case 'at':
        datepicker.label = 'Am';
        datepicker.range = false;
        break;
      case 'after':
        datepicker.label = 'Ab';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').shift();
        }
        break;
      case 'before':
        datepicker.label = 'Bis';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').pop();
        }
        break;
      case 'range':
        datepicker.range = true;
        datepicker.label = 'Zeitraum';
        break;
    }
    group.value = target.value;
  };

  today.addEventListener('click', todayClickHandler);
  thisWeek.addEventListener('click', thisWeekClickHandler);
  lastWeek.addEventListener('click', lastWeekClickHandler);
  group.addEventListener('checkedChange', checkedChangeHandler);

  return () => {
    today.removeEventListener('click', todayClickHandler);
    thisWeek.removeEventListener('click', thisWeekClickHandler);
    lastWeek.removeEventListener('click', lastWeekClickHandler);
    group.removeEventListener('checkedChange', checkedChangeHandler);
  };
}
