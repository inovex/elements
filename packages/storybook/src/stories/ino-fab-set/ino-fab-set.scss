@import '../utils';

// only selects Stories of ino-fab-set without decorating stories with wrapper classes
.sbdocs.sbdocs-content:has(#anchor--buttons-ino-fab-set--default) {
  [scale] {
    width: 100%;
  }
}

@include story-container {
  min-width: 10px;
  min-height: 10px;
}

.ino-fab-set--bottom-left {
  position: absolute;
  bottom: 20px;
  left: 20px;
}
