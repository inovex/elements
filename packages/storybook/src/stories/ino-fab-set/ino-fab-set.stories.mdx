import { Meta, Description } from "@storybook/addon-docs";
import readme from "../../../../elements/src/components/ino-fab-set/readme.md";

<Meta title="Buttons/ino-fab-set/Documentation" />

<Description markdown={readme} />
