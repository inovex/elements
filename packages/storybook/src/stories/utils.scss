/**
 * Applies styling to the container of a story.
 * The styling is encapsulated within this mixin because the ID may change in the future.
 */
@mixin story-container {
  #root-inner {
    @content;
  }
}
