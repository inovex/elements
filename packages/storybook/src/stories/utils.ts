import { html } from 'lit-html';
import ICONS from '../../../elements/src/components/ino-icon/icons';

export const maybeCreateStoryArgs = (story: any) => {
  if (!story.args) {
    story.args = {};
  }
  if (!story.argTypes) {
    story.argTypes = {};
  }
};

export const withIconControl = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: getIcons(),
  };
};

export const withColorScheme = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: [
      'primary',
      'secondary',
      'success',
      'warning',
      'error',
      'light',
      'dark',
    ],
  };
};

export const withSortDirection = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    options: ['desc', 'asc'],
    control: {
      type: 'select',
    },
  };
};

export const getIcons = () => ICONS.filter((icon) => !icon.startsWith('_'));

export const defaultDecorator = (story, className?: string) => {
  return html` <div class="${className ?? ''}">${story()}</div> `;
};
