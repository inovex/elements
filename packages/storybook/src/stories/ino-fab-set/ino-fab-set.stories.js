import { boolean, select, text } from '@storybook/addon-knobs';
import withStencilReadme from '_local-storybookcore/with-stencil-readme';

import componentReadme from '_local-elements/src/components/ino-fab-set/readme.md';

import ICONS from '_local-elements/src/components/ino-icon/icons';
import './ino-fab-set.scss';

const tooltipPlacementOptions = [
  'top',
  'top-start',
  'top-end',
  'right',
  'right-start',
  'right-end',
  'bottom',
  'bottom-start',
  'bottom-end',
  'left',
  'left-start',
  'left-end',
  'none'
];

// https://github.com/storybooks/storybook/issues/4337#issuecomment-428495664
function subscribeToComponentEvents() {
  // == event block
  const eventHandler = function(e) {
    const el = e.target;

    if (el.tagName !== 'INO-FAB-SET') return;

    const newState =
      el.getAttribute('ino-open-dial') === 'false' ? 'true' : 'false';
    el.setAttribute('ino-open-dial', newState);
  };

  document.addEventListener('click', eventHandler);
  // == event block

  // unsubscribe function will be called by Storybook
  return () => {
    document.removeEventListener('click', eventHandler);
  };
}

export default {
  title: 'Buttons/<ino-fab-set>',
  parameters: {
    actions: {
      handles: ['click ino-fab']
    }
  },
  decorators: [
    withStencilReadme(componentReadme),
    story => {
      subscribeToComponentEvents();
      return story();
    }
  ]
};

export const DefaultUsage = () => /*html*/ `
    <div class="story-fab-set">
      <ino-fab-set
      ino-open-dial="${boolean('ino-open-dial', false)}"
      ino-top-bottom-location="${select(
        'ino-top-bottom-location',
        ['top', 'bottom'],
        'bottom',
        'FAB-set'
      )}"
      ino-left-right-location="${select(
        'in-left-right-location',
        ['left', 'right'],
        'left',
        'FAB-set'
      )}"
      ino-dial-direction="${select(
        'ino-dial-direction',
        ['top', 'bottom', 'left', 'right'],
        'top',
        'FAB-set'
      )}">

        <ino-fab
          ino-icon="${select('ino-icon-fab1', ICONS, 'star', 'First FAB')}"
          ino-label="${text('ino-label-fab1', 'Label 1', 'First FAB')}"
          ino-mini="${boolean('ino-mini-fab1', true, 'First FAB')}"
        ></ino-fab>
        <ino-fab
          ino-icon="${select('ino-icon-fab2', ICONS, 'favorite', 'Second FAB')}"
          ino-label="${text('ino-label-fab2', 'Label 2', 'Second FAB')}"
          ino-mini="${boolean('ino-mini-fab2', true, 'Second FAB')}"
        ></ino-fab>
        <ino-fab
          ino-icon="${select('ino-icon-fab3', ICONS, 'info', 'Third FAB')}"
          ino-label="${text('ino-label-fab3', 'Label 3', 'Third FAB')}"
          ino-mini="${boolean('ino-mini-fab3', true, 'Third FAB')}"
        ></ino-fab>
      </ino-fab-set>
    </div>
  `;
