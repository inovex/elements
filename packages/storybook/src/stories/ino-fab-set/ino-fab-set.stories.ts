import { Components } from '@inovex.de/elements';
import { useEffect } from '@storybook/client-api';
import { Story } from '@storybook/web-components';
import { html } from 'lit-html';
import { defaultDecorator } from '../utils';

import './ino-fab-set.scss';

// https://github.com/storybooks/storybook/issues/4337#issuecomment-428495664
function subscribeToComponentEvents() {
  // == event block
  const eventHandler = function (e) {
    const el = e.target;

    if (el.tagName !== 'INO-FAB-SET') {
      return;
    }

    const newState =
      el.getAttribute('ino-open-dial') === 'false' ? 'true' : 'false';
    el.setAttribute('ino-open-dial', newState);
  };

  document.addEventListener('click', eventHandler);
  // == event block

  // unsubscribe function will be called by Storybook
  return () => {
    document.removeEventListener('click', eventHandler);
  };
}

export default {
  title: 'Buttons/<ino-fab-set>',
  component: 'ino-fab-set',
  parameters: {
    actions: {
      handles: ['click ino-fab'],
    },
  },
  decorators: [
    defaultDecorator,
    (story) => {
      useEffect(() => {
        const clickHandler = function (e) {
          if (e.target.tagName !== 'INO-FAB-SET') {
            return;
          }
          const el = e.target as HTMLInoFabSetElement;
          el.openDial = !el.openDial;
        };

        document.addEventListener('click', clickHandler);
        return () => {
          document.removeEventListener('click', clickHandler);
        };
      });

      return story();
    },
  ],
};

export const Playground: Story<Components.InoFabSet> = (args) => html`
  <ino-fab-set
    open-dial="${args.openDial}"
    top-bottom-location="${args.topBottomLocation}"
    left-right-location="${args.leftRightLocation}"
    dial-direction="${args.dialDirection}"
  >
    <ino-fab label="First FAB">
      <ino-icon slot="icon-leading" icon="star"></ino-icon>
    </ino-fab>
    <ino-fab label="Second FAB">
      <ino-icon slot="icon-leading" icon="favorite"></ino-icon>
    </ino-fab>
    <ino-fab label="Third FAB">
      <ino-icon slot="icon-leading" icon="info"></ino-icon>
    </ino-fab>
  </ino-fab-set>
`;

Playground.args = {
  openDial: false,
  dialDirection: 'top',
  leftRightLocation: 'left',
  topBottomLocation: 'bottom',
};

const VERTICAL_POSITION_OPTIONS = ['top', 'bottom'];
const HORIZONTAL_POSITION_OPTIONS = ['left', 'right'];

Playground.argTypes = {
  topBottomLocation: {
    control: {
      type: 'select',
    },
    options: VERTICAL_POSITION_OPTIONS,
  },
  leftRightLocation: {
    control: {
      type: 'select',
    },
    options: HORIZONTAL_POSITION_OPTIONS,
  },
  dialDirection: {
    control: {
      type: 'select',
    },
    options: [...VERTICAL_POSITION_OPTIONS, ...HORIZONTAL_POSITION_OPTIONS],
  },
};
