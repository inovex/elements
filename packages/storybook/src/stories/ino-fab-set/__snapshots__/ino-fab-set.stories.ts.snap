// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons/<ino-fab-set> Playground smoke-test 1`] = `
<div id="root-inner">
  <div class>
    <div style="position: absolute">
      <ino-fab-set open-dial="false"
                   top-bottom-location="bottom"
                   left-right-location="left"
                   dial-direction="top"
      >
        <ino-icon slot="icon-opened"
                  icon="_fab_set_arrow_down"
        >
        </ino-icon>
        <ino-icon slot="icon-closed"
                  icon="_fab_set_arrow_up"
        >
        </ino-icon>
        <ino-fab label="First FAB">
          <ino-icon slot="icon-leading"
                    icon="star"
          >
          </ino-icon>
        </ino-fab>
        <ino-fab label="Second FAB">
          <ino-icon slot="icon-leading"
                    icon="favorite"
          >
          </ino-icon>
        </ino-fab>
        <ino-fab label="Third FAB">
          <ino-icon slot="icon-leading"
                    icon="info"
          >
          </ino-icon>
        </ino-fab>
      </ino-fab-set>
    </div>
  </div>
</div>
`;
